app/assets/javascripts/application.js.coffee

```diff
@@ -10,4 +10,13 @@
 #= require spectrum
 #= require_tree ./components
 #= require_tree ./pages
-#= require_self
\ No newline at end of file
+#= require_self
+
+format = (icon) ->
+  originalOption = icon.element
+  '<i class="fa ' + $(originalOption).data('icon') + '"></i> ' + icon.text
+
+$(document).ready ->
+  $('.js-example-basic-single').select2
```

Comment:
How about giving that class a more meaningful name like `select2-fountawesome-icon`?